### PR TITLE
Update Vault APIs to v2 (PAN-16801)

### PR DESCRIPTION
**Pull request by kenany**
- [x] Update paths from `v2beta` to `v2`.
- [x] Vault v2 is in LVE.
- [x] Revert CI to run on LVE.

---

Introduced lombok to autogenerate builders, getters, and setters.

### CODE REVIEW

File: packages/pangea-sdk/.sdk-ci.yml [hunk 1]
```diff
@@ -26,7 +26,7 @@ sdk-integration-tests:
     SERVICE_FILE_INTEL_ENV: LVE
     SERVICE_USER_INTEL_ENV: LVE
     SERVICE_REDACT_ENV: LVE
-    SERVICE_VAULT_ENV: LVE
+    SERVICE_VAULT_ENV: DEV
```
Comment (kenany) [inline packages/pangea-sdk/.sdk-ci.yml:29]: Undo this when Vault v2 is in LVE.
```suggestion
    SERVICE_VAULT_ENV: LVE
```

File: packages/pangea-sdk/src/main/java/cloud/pangeacyber/pangea/authn/requests/UserListRequest.java [hunk 2]
```diff
@@ -54,6 +54,7 @@ public Builder() {}
 		/**
 		 * @deprecated Use setFilter(FilterUserList filter) instead
 		 */
+		@Deprecated
 		public Builder setFilter(Filter filter) {
```
Comment (kenany) [inline packages/pangea-sdk/src/main/java/cloud/pangeacyber/pangea/authn/requests/UserListRequest.java:58]: If we're doing a major version bump, consider removing this instead.

**Review by pangea-andrest — DISMISSED** [review]

File: packages/pangea-sdk/src/main/java/cloud/pangeacyber/pangea/vault/results/StateChangeResult.java [hunk 3]
```diff
@@ -1,38 +1,42 @@
...
+import lombok.NoArgsConstructor;
 
+@Data
 @JsonIgnoreProperties(ignoreUnknown = true)
+@NoArgsConstructor
 public class StateChangeResult {
 
-	@JsonProperty("id")
-	String id;
+	@JsonProperty("created_at")
```
Comment (pangea-andrest) [inline packages/pangea-sdk/src/main/java/cloud/pangeacyber/pangea/vault/results/StateChangeResult.java:16]: Missing docs?

File: packages/pangea-sdk/src/main/java/cloud/pangeacyber/pangea/vault/results/SymmetricGenerateResult.java [hunk 4]
```diff
@@ -1,26 +1,26 @@
...
+@EqualsAndHashCode(callSuper = true)
 @JsonIgnoreProperties(ignoreUnknown = true)
+@NoArgsConstructor
 public class SymmetricGenerateResult extends CommonGenerateResult {
 
+	@NonNull
+	@JsonProperty("type")
+	ItemType type = ItemType.SYMMETRIC_KEY;
+
 	@JsonProperty("algorithm")
```
Comment (pangea-andrest) [inline packages/pangea-sdk/src/main/java/cloud/pangeacyber/pangea/vault/results/SymmetricGenerateResult.java:21]: Same here and in some others classes.

File: CHANGELOG.md [hunk 5]
```diff
@@ -12,6 +12,15 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 - Detect-only Redact for Sanitize.
 - Support for `domains` field in `v2/user/breached` endpoint in User Intel service
 
+### Changed
+
+- Vault APIs have been updated to v2.
+
+### Removed
+
+- Deprecated functions `UserListRequest.Builder.setFilter(Filter filter)` and
```
Comment (pangea-andrest) [inline CHANGELOG.md:21]: nit: would be nice to indicate which service these classes are from

**Review by pangea-andrest — APPROVED** [review]

File: packages/pangea-sdk/src/main/java/cloud/pangeacyber/pangea/vault/results/StateChangeResult.java (hunk 3 above)
Comment (kenany) [reply packages/pangea-sdk/src/main/java/cloud/pangeacyber/pangea/vault/results/StateChangeResult.java:17]: Added more.

File: packages/pangea-sdk/src/main/java/cloud/pangeacyber/pangea/vault/results/SymmetricGenerateResult.java (hunk 4 above)
Comment (kenany) [reply packages/pangea-sdk/src/main/java/cloud/pangeacyber/pangea/vault/results/SymmetricGenerateResult.java:24]: Added more.

File: CHANGELOG.md (hunk 5 above)
Comment (kenany) [reply CHANGELOG.md]: Sure, done.

**Review by pangea-andrest — APPROVED** [review]

**Pull request merged**